src/awscdk-construct.ts

```
@@ -119,32 +131,49 @@ export class AwsCdkConstructLibrary extends ConstructLibrary {
     if (!options.cdkVersion) {
       throw new Error('Required field cdkVersion is not specified.');
     }
-
     this.cdkVersion = options.cdkVersionPinning ? options.cdkVersion : `^${options.cdkVersion}`;
+    const cdkMajorVersion = semver.minVersion(this.cdkVersion)?.major ?? 2;
+
     this.cdkDependenciesAsDeps = options.cdkDependenciesAsDeps ?? true;
 
-    const cdkMajorVersion = semver.minVersion(this.cdkVersion)?.major ?? 1;
     if (options.constructsVersion) {
+      const constructsMajorVersion = semver.minVersion(options.constructsVersion)?.major;
+      if (constructsMajorVersion !== undefined &&
+        constructsMajorVersion < 10 && cdkMajorVersion === 2) {
+        throw new Error('CDK 2.x requires constructs version >= 10');
+      }
       this.addPeerDeps(`constructs@^${options.constructsVersion}`);
       this.addDevDeps(`constructs@${options.constructsVersion}`);
     } else if (cdkMajorVersion === 1) {
+      if (options.cdkAssert) {
+        this.addDevDeps(this.formatModuleSpec('@aws-cdk/assert'));
```

addPeerDeps automatically adds a dev dependency, so I don't think this is needed

---

It looks like we have been explicitly setting that to `false`

https://github.com/projen/projen/blob/dcfcfde305885778b804abe624f0f000b177a3de/src/awscdk-construct.ts#L127

---

Oh right because it is explicitly taken cared of in addCdkDependency(). 

I am wondering what's the utility of add addCdkDependency() for v2... maybe we should throw an error there?

---

I think it is still used for the `-alpha` modules.

---

Yes you are right.

I do think that we can enable the pinned dev dependency option when using v2

---

Ok i'll update that.

---

Ok I reworked this a little. I kept the v1 functionality the same, but for v2 `pinDevDependency` will not be set so will default to `true`. Also, `cdkDependencies` will not apply for v2. You will just add any `alpha` modules as `peerDeps` and they will automatically get set as `devDeps` as well due to `pinDevDependency`.